packages/ember/tests/helpers/helper_registration_test.js

```diff
@@ -1,112 +1,67 @@
+import { moduleFor, ApplicationTestCase } from 'internal-test-helpers';
 import { Controller, Service, inject } from 'ember-runtime';
-import { run } from 'ember-metal';
-import { compile } from 'ember-template-compiler';
-import {
-  Helper,
-  helper,
-  setTemplates,
-  setTemplate
-} from 'ember-glimmer';
-import { Application } from 'ember-application';
-import { Router } from 'ember-routing';
-import { jQuery } from 'ember-views';
-
-let App, appInstance;
-
-QUnit.module('Application Lifecycle - Helper Registration', {
-  teardown() {
-    run(() => {
-      if (App) {
-        App.destroy();
-      }
+import { Helper, helper } from 'ember-glimmer';
 
-      App = appInstance = null;
-      setTemplates({});
-    });
-  }
-});
+moduleFor('Application Lifecycle - Helper Registration', class extends ApplicationTestCase {
+  ['@test Unbound dashed helpers registered on the container can be late-invoked'](assert) {
+    this.addTemplate('application', `<div id='wrapper'>{{x-borf}} {{x-borf 'YES'}}</div>`);
 
-function boot(callback) {
-  run(() => {
-    App = Application.create({
-      name: 'App',
-      rootElement: '#qunit-fixture'
-    });
-
-    App.deferReadiness();
+    let myHelper = helper(params => params[0] || 'BORF');
+    this.add('helper:x-borf', myHelper);
```

Comment:
As this test is looking to test helpers "registered on the container" `helper:x-borf` should be registered here instead of added to the resolver.

```
this.application.register('helper:x-borf', myHelper);
```

Instead of adding.